langserver/build_context.go

```diff
@@ -36,17 +36,9 @@ func (h *LangHandler) defaultBuildContext() *build.Context {
 }
 
 func (h *HandlerShared) OverlayBuildContext(ctx context.Context, orig *build.Context, useOSFileSystem bool) *build.Context {
-	// Copy to avoid a race condition if concurrent goroutines
-	// bind new mountpoints to the ctxvfs.NameSpace. This lets us lock
-	// once here, instead of once per call to any ctxvfs.FileSystem
-	// method.
 	h.Mu.Lock()
-	fs := make(ctxvfs.NameSpace, len(h.FS))
-	for path, mounts := range h.FS {
-		fs[path] = mounts
-	}
+	fs := h.FS
 	h.Mu.Unlock()
```

Comment:
we still keep this mutex due to potential race conditions to the pointer h.FS

at a later stage I'll clean up all this ceremony around needing to protect h's fields due to calls to `Reset`. But that is unrelated to this PR